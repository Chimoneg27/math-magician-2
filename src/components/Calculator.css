@import url('https://fonts.googleapis.com/css2?family=Montserrat&family=Poppins&family=Roboto&display=swap');

* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.calculator {
  margin: 5rem auto;
  padding: 0;
  width: 500px;
  font-family: 'Montserrat', 'Times New Roman', Times, serif;
}

.outputs {
  width: 100%;
  font-size: 1.25rem;
  color: white;
  background-color: grey;
  padding: 1rem;
  text-align: end;
}

.row1 {
  display: flex;
  justify-content: space-evenly;
  gap: 0;
}

.row1 button {
  width: 125px;
  padding: 0.75rem;
  font-weight: bold;
  font-size: 1rem;
  border: 1px solid rgb(156, 123, 123);
}

.far-right {
  background-color: rgb(255, 115, 0);
}

.rows {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.bottom {
  display: flex;
  font-size: 1rem;
  width: 100%;
  font-weight: bold;
  border: 1px solid rgb(156, 123, 123);
}

.zero {
  width: 50%;
  padding: 0.75rem;
  font-weight: bold;
}

.point {
  width: 25%;
  padding: 0.75rem;
  font-weight: bold;
}

.bottom .far-right {
  width: 25%;
  font-size: 1rem;
  border: 1px solid rgb(156, 123, 123);
  font-weight: bold;
}

.container {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 1rem;
}
