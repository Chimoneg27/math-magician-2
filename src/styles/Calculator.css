@import url('https://fonts.googleapis.com/css2?family=Montserrat&family=Poppins&family=Roboto&display=swap');

* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

:root {
  --text: #01130b;
  --background: #fafffd;
  --primary: #7bf9be;
  --secondary: #a3ffd1;
  --accent: #0bcd72;
}

.calculator {
  width: 60%;
  padding: 0;
  font-family: 'Montserrat', 'Times New Roman', Times, serif;
}

.outputs {
  width: 100%;
  font-size: 1.25rem;
  color: var(--secondary);
  background-color: var(--text);
  padding: 1rem;
  text-align: end;
}

.row1 {
  display: flex;
  justify-content: space-evenly;
  gap: 0;
}

.row1 button {
  width: 33%;
  padding: 0.75rem;
  font-weight: bold;
  font-size: 1rem;
  border: 1px solid rgb(156, 123, 123);
}

.far-right {
  background-color: var(--accent);
}

.rows {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.bottom {
  display: flex;
  font-size: 1rem;
  width: 100%;
  font-weight: bold;
  border: 1px solid rgb(156, 123, 123);
}

.zero {
  width: 50%;
  padding: 0.75rem;
  font-weight: bold;
}

.point {
  width: 25%;
  padding: 0.75rem;
  font-weight: bold;
}

.bottom .far-right {
  width: 25%;
  font-size: 1rem;
  border: 1px solid rgb(156, 123, 123);
  font-weight: bold;
}

.container {
  width: 100%;
  display: flex;
  align-items: flex-start;
  justify-content: space-evenly;
  padding: 1rem;
  margin-top: 2rem;
}

.text {
  padding: 1rem;
  font-size: 1.25rem;
  font-family: "Roboto", 'Times New Roman', Times, serif;
  width: 40%;
}
