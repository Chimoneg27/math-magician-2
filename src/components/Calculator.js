import React, { useState } from 'react';
import '../styles/Calculator.css';
import Buttons from './button';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorState, setCalculatorState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newCalculatorState = calculate(calculatorState, buttonName);
    setCalculatorState(newCalculatorState);
  };

  return (
    <div className="container">
      <div className="text">
        <h1>Lets Do Some Math!</h1>
      </div>
      <div className="calculator">
        <div className="outputs">
          {calculatorState.next || calculatorState.total || '0'}
        </div>
        <div className="keys">
          <Buttons clickHandler={handleButtonClick} />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
