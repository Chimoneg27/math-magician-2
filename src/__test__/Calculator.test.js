import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator Component Snapshot Tests', () => {
  test('displays the Calculator component', () => {
    const component = renderer.create(<Calculator />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
