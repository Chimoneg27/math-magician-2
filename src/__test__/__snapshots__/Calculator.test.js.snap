// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component Snapshot Tests displays the Calculator component 1`] = `
<div
  className="container"
>
  <div
    className="text"
  >
    <h1>
      Lets Do Some Math!
    </h1>
  </div>
  <div
    className="calculator"
  >
    <div
      className="outputs"
    >
      0
    </div>
    <div
      className="keys"
    >
      <div
        className="rows"
      >
        <div
          className="row1"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="far-right"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="row1"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="far-right"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
        <div
          className="row1"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="far-right"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="row1"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="far-right"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="bottom"
        >
          <button
            className="zero"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="point"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="far-right"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
